govcd/adminvdc_nsxt_test.go

```diff
@@ -121,6 +121,7 @@ func (vcd *TestVCD) Test_CreateNsxtOrgVdc(check *C) {
 		check.Assert(vdc.Vdc.Name, Equals, vdcConfiguration.Name)
 		check.Assert(vdc.Vdc.IsEnabled, Equals, vdcConfiguration.IsEnabled)
 		check.Assert(vdc.Vdc.AllocationModel, Equals, vdcConfiguration.AllocationModel)
+		check.Assert(len(adminVdc.AdminVdc.ResourcePoolRefs) > 0, Equals, true)
```

Comment:
This structure change broke marshaling in the test.

```
START: adminvdc_nsxt_test.go:19: TestVCD.Test_CreateNsxtOrgVdc
adminvdc_nsxt_test.go:129:
    check.Assert(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"error updating VDC: API Error: 400: [ 596b4414-bd0a-4aff-afd0-ed4dc2cb9a9f ] HTTP 400 Bad Request\n - Undeclared namespace prefix \"vmext\"\n at [row,col {unknown-source}]: [100,30]"} ("error updating VDC: API Error: 400: [ 596b4414-bd0a-4aff-afd0-ed4dc2cb9a9f ] HTTP 400 Bad Request\n - Undeclared namespace prefix \"vmext\"\n at [row,col {unknown-source}]: [100,30]")

FAIL: adminvdc_nsxt_test.go:19: TestVCD.Test_CreateNsxtOrgVdc
```


Do you have any chance to run the test on your env or would you need help?

I think the bug comes out in a very simple scenario which could even be created outside of tests I guess:

Triggering `adminVdc.Update()` with the structure received throws a marshaling error.

Reply:
I push another commit to overcome above issue. 
First I used the main branch to create a test on the expected XML for marshaling of AdminVdc. Then I moved the test to the PR and made code changes to satisfy the test.
Just to be clear I did not do a real integration test but guaranteed that the marshaling of AdminVdc is not impacted by this PR.